src/modules/cpu/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,1814 @@
+                            Rpp32s rowIndex = k * 3;
+
+                            pTemp[0] = _mm256_mul_ps(pRow[rowIndex], pFilter[filterIndex]);
+                            pTemp[1] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 7), avx_pxMaskRotate0To3), pFilter[filterIndex + 1]);
+                            pTemp[2] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex], pRow[rowIndex + 1], 63), avx_pxMaskRotate0To6), pFilter[filterIndex + 2]);
+                            pDst[0] = _mm256_add_ps(pDst[0], _mm256_add_ps(_mm256_add_ps(pTemp[0], pTemp[1]), pTemp[2]));
+
+                            pTemp[0] = _mm256_mul_ps(pRow[rowIndex + 1], pFilter[filterIndex]);
+                            pTemp[1] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex + 1], pRow[rowIndex + 2], 7), avx_pxMaskRotate0To3), pFilter[filterIndex + 1]);
+                            pTemp[2] = _mm256_mul_ps(_mm256_permutevar8x32_ps(_mm256_blend_ps(pRow[rowIndex + 1], pRow[rowIndex + 2], 63), avx_pxMaskRotate0To6), pFilter[filterIndex + 2]);
+                            pDst[1] = _mm256_add_ps(pDst[1], _mm256_add_ps(_mm256_add_ps(pTemp[0], pTemp[1]), pTemp[2]));
+                        }
+
+                        increment_row_ptrs(srcPtrTemp, kernelSize, 16);
+                        rpp_store_gaussian_filter_3x3_host(dstPtrTemp, pDst);
+                        dstPtrTemp += 16;
+                    }
+    #endif
+                    vectorLoopCount += padLength * 3;
+                    for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                    {
+                        gaussian_filter_generic_tensor(srcPtrTemp, dstPtrTemp, vectorLoopCount / 3, kernelSize, padLength, unpaddedWidth, rowKernelLoopLimit, filterTensor, 3);
+                        increment_row_ptrs(srcPtrTemp, kernelSize, 1);
+                        dstPtrTemp++;
+                    }
+                    // for the first padLength rows, we need not increment the src row pointers to next rows
+                    increment_row_ptrs(srcPtrRow, kernelSize, (!padLengthRows) ? srcDescPtr->strides.hStride : 0);
+                    dstPtrRow += dstDescPtr->strides.hStride;
+                }
+            }
+            else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+            {
+                /* exclude ((2 * padLength) * 3) number of columns from alignedLength calculation
+                    since (padLength * 3) number of columns from the beginning and end of each row will be computed using raw c code */
+                Rpp32u alignedLength = ((bufferLength - (2 * padLength) * 3) / 24) * 24;
+#if __AVX2__
+                __m256 pFilter[9];
+                for (int i = 0; i < 9; i++)
+                    pFilter[i] = _mm256_set1_ps(filterTensor[i]);
+#endif
+                T *dstPtrChannels[3];
```

Comment:
Since this filter part is common for all layout variants in kernel size 3
you can move this outside after L320 and add it there and remove this code inside layout variants

Reply:
Done